AdminWebsite/AdminWebsite/ClientApp/src/app/booking/hearing-schedule/date-error-messages/date-error-messages.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { DateErrorMessagesComponent } from './date-error-messages';
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { DebugElement, ElementRef } from '@angular/core';
+import { By } from '@angular/platform-browser';
+
+describe('Date error message test suite', () => {
+    let component: DateErrorMessagesComponent;
+    let fixture: ComponentFixture<DateErrorMessagesComponent>;
+    let debugElement: DebugElement;
+    let requiredMessage: ElementRef;
+    let pastMessage: ElementRef;
+    let weekendHolidayMessage: ElementRef;
+
+    beforeEach(
+        waitForAsync(() => {
+            TestBed.configureTestingModule({
+                declarations: [DateErrorMessagesComponent]
+            }).compileComponents();
+        })
+    );
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(DateErrorMessagesComponent);
+        component = fixture.componentInstance;
+
+        debugElement = fixture.debugElement;
+    });
+
+    it('should show Select a date error message when no date is selected', () => {
+        component.required = true;
+        fixture.detectChanges();
+        requiredMessage = debugElement.query(By.css('#hearingDate-required-error'));
+        expect(requiredMessage.nativeElement.innerText).toEqual('Select a date');
```

Comment:
Could we change the assertions to check if the required message element is present. We are already querying that using the id so we don't have to check the string content